core/cas-server-core-logging-api/src/test/resources/log4j2-test.xml

```diff
@@ -16,6 +16,23 @@
             </Policies>
         </RollingFile>
 
+        <!-- Required by CasLoggerFactoryTests. Filename is linked to the test. -->
+        <!-- Setting mmediateFlush so test can check contents -->
+        <RollingFile name="stacktracefile" fileName="build/slf4j-exceptions.log" append="true"
+                     filePattern="build/slf4j-%d{yyyy-MM-dd-HH}-%i.log.gz" bufferedIO="false" immediateFlush="true">
+            <PatternLayout pattern="%highlight{%d %p [%c] - %m%n}" />
+            <Policies>
+                <OnStartupTriggeringPolicy />
+                <SizeBasedTriggeringPolicy size="10 MB"/>
+                <TimeBasedTriggeringPolicy />
+            </Policies>
+        </RollingFile>
+
+        <CasAppender name="CasStackTraceFile">
+            <AppenderRef ref="stacktracefile" />
+            <ExceptionOnlyFilter/>
+        </CasAppender>
+
```

Comment:
You should also update the default log4j2.xml file that is in the resources module I believe. That would be the file that kicks in, if the overlay does not present its own log4j2 config file in the properties.

Reply:
That log4j2.xml in resources is already updated with stacktracefile appender that has the ExceptionOnlyFilter. I made the other changes as well.